.github/workflows/ci.yml

```
@@ -0,0 +1,192 @@
+name: Build and test
+
+# The main C++ CI is split into three steps: lint, build, and system tests. The
+# linting and building are independent but we make the build wait so as not to
+# waste resources compiling and testing a commit which doesn't pass the linter.
+#
+# System tests are slow and need input data. To save caching, network, and job
+# time we only run the full system tests on Ubuntu (although this can easily be
+# changed).
+
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
```

```suggestion
  pull_request:
    branches: ["main"]
```
Do we not want the tests to run on PRs that target `main`? Or am I just not up to scratch with my GH actions syntax & the `push` field takes care of this?

---

This is: on all pushes to `main`¹, all pull requests, and if a person clicks the button in the actions tab.

 ----
¹ which only happens on PR merge... but we might've allowed direct push to main if we were a smaller team or whatever.

---

Your way would mean a PR against a PR would _not_ run the CI. (Perhaps we want it like that?)

---

Probably want PRs against PRs to run the system tests (although barring the present chain, we're trying to avoid this anyway, so should be a non-issue). I think it's all good as you have it here :+1: